core/src/main/java/hudson/scm/SCMDescriptor.java

```diff
@@ -56,7 +62,8 @@
      * @deprecated No longer used by default.
      */
     @Deprecated
-    public volatile int generation = 1;
+    @Restricted(NoExternalUse.class) @RestrictedSince("TODO")
+    public volatile AtomicInteger generation = new AtomicInteger(1);
```

Comment:
Making a variable atomic and volatile is redundant. The AtomicInteger (or any other AtomicReference) can be final as the volatile field is contained in those classes.

Reply:
Thanks, I changed it.